addons/html_editor/static/src/editor/table/table_plugin.js

```diff
@@ -284,15 +283,110 @@ export class TablePlugin extends Plugin {
         table.remove();
         this.shared.setCursorStart(p);
     }
-    deleteBackwardBefore({ targetNode, targetOffset }) {
-        // If the cursor is at the beginning of a row, prevent deletion.
-        if (targetNode.nodeType === Node.ELEMENT_NODE && isRow(targetNode) && !targetOffset) {
-            return true;
+
+    // @todo @phoenix: handle deleteBackward on table cells
+    // deleteBackwardBefore({ targetNode, targetOffset }) {
+    //     // If the cursor is at the beginning of a row, prevent deletion.
+    //     if (targetNode.nodeType === Node.ELEMENT_NODE && isRow(targetNode) && !targetOffset) {
+    //         return true;
+    //     }
+    // }
+
+    /**
+     * Removes fully selected rows or columns, clears the content of selected
+     * cells otherwise.
+     *
+     * @param {NodeListOf<HTMLTableCellElement>} selectedTds - Non-empty
+     * NodeList of selected table cells.
+     */
+    deleteTableCells(selectedTds) {
+        const rows = [...closestElement(selectedTds[0], "tr").parentElement.children].filter(
+            (child) => child.nodeName === "TR"
+        );
+        const firstRowCells = [...rows[0].children].filter(
+            (child) => child.nodeName === "TD" || child.nodeName === "TH"
+        );
+        const firstCellRowIndex = getRowIndex(selectedTds[0]);
+        const firstCellColumnIndex = getColumnIndex(selectedTds[0]);
+        const lastCellRowIndex = getRowIndex(selectedTds[selectedTds.length - 1]);
+        const lastCellColumnIndex = getColumnIndex(selectedTds[selectedTds.length - 1]);
+
+        const areFullColumnsSelected =
+            firstCellRowIndex === 0 && lastCellRowIndex === rows.length - 1;
+        const areFullRowsSelected =
+            firstCellColumnIndex === 0 && lastCellColumnIndex === firstRowCells.length - 1;
+
+        if (areFullColumnsSelected) {
+            for (let index = firstCellColumnIndex; index <= lastCellColumnIndex; index++) {
+                this.removeColumn({ cell: firstRowCells[index] });
+            }
+            return;
+        }
+
+        if (areFullRowsSelected) {
+            for (let index = firstCellRowIndex; index <= lastCellRowIndex; index++) {
+                this.removeRow({ row: rows[index] });
```

Comment:
Remove row has an error: `this.shared.setCursorStart({ anchorNode: siblingRow, anchorOffset: 0 })` (it should take a node, not an Object as parameter)